addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -237,8 +237,10 @@ def _create_invoices(self, sale_orders):
                             (product_lines, 'price_total', -1),
                             (tax_lines, 'amount_currency', 1),
                         ):
-                            amt = delta_amount / len(lines)
-                            for line in lines:
+                            remaining = delta_amount
+                            for i, line in list(enumerate(lines))[::-1]:
+                                amt = order.currency_id.round(remaining / (i+1))
+                                remaining -= amt
```

Comment:
This looks difficult to read. As it's main use is to push the remaining 'indivisible' rounding on the last line
We could put it on the first line instead
```diff
-                            for i, line in list(enumerate(lines))[::-1]:
-                                amt = order.currency_id.round(remaining / (i+1))
+                            for line in lines:
+                                amt = order.currency_id.round(remaining / len(lines))
+                                if order.currency_id.is_zero(amt):
+                                    amt = remaining
```